### PR TITLE
Stm32u5 support

### PR DESCRIPTION
**Pull request by jcamdr**
## Detailed description
Addition of the documentation reference for the STM32U5.
Addition of the DP_TARGETIDR id for the STM32U5.
Addition of the stm32l4_device_info for the STM32U5.
Addition of the RAM map for the STM32U5.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: src/target/stm32l4.c [hunk 1]
```diff
@@ -583,7 +622,7 @@ static bool stm32l4_attach(target_s *const t)
 	/* Free any previously built memory map */
 	target_mem_map_free(t);
 	/* And rebuild the RAM map */
-	if (device->family == STM32L4_FAMILY_L55x)
+	if ((device->family == STM32L4_FAMILY_L55x) || (device->family == STM32L4_FAMILY_U5xx))
```
Comment (dragonmux) [inline src/target/stm32l4.c:625]: The parens are unnecessary and suppress a GCC warning about accidental assignment, please drop them:
```suggestion
	if (device->family == STM32L4_FAMILY_L55x || device->family == STM32L4_FAMILY_U5xx)
```

**Review by dragonmux — CHANGES_REQUESTED** [review]
This is looking great, there's just one tweak that needs making and when done we can then merge this. Thank you for the contribution!

**Comment by jcamdr** [comment]
I will agree if you ask me to integrate the change request into the second commit. I was not so confident that the operation proposed by the github interface will do that or not.

**Comment by dragonmux** [comment]
Yes, please do - sorry, meant to get back to this PR earlier and ask you to do just that. With that change made part of the second commit, we're happy to mark as accepted and merge this

**Comment by jcamdr** [comment]
Don't worry, it's part of the normal learning curve for working together.

**Review by dragonmux — APPROVED** [review]
LGTM now, merging.

**Comment by jcamdr** [comment]
Thank you!
I will now think about the flash, the biggest part.

**Pull request merged**